screen_brightness_control/__init__.py

```diff
@@ -392,7 +410,60 @@ def fade_brightness(
         interval: float = 0.01,
         increment: int = 1,
         force: bool = False,
-        logarithmic: bool = True
+        logarithmic: bool = True,
+        blocking: bool = True,
+        stoppable: bool = True
+    ) -> IntPercentage:
+        '''
+        Gradually change the brightness of this display to a set value.
+        Can execute in the current thread, blocking until completion,
+        or in a separate thread, allowing concurrent operations.
+        When set as non-blocking and stoppable, a new fade can halt the this operation.
+
+        Args:
+            finish (.types.Percentage): the brightness level to end up on
+            start (.types.Percentage): where the fade should start from. Defaults
+                to whatever the current brightness level for the display is
+            interval: time delay between each change in brightness
+            increment: amount to change the brightness by each time (as a percentage)
+            force: [*Linux only*] allow the brightness to be set to 0. By default,
+                brightness values will never be set lower than 1, since setting them to 0
+                often turns off the backlight
+            logarithmic: follow a logarithmic curve when setting brightness values.
+                See `logarithmic_range` for rationale
+            blocking: run this function in the current thread and block until it completes
+            stoppable: whether this fade will be stopped by starting a new fade on the same display
+
+        Returns:
+            The brightness of the display after the fade is complete.
+            See `.types.IntPercentage`
+
+            .. warning:: Deprecated
+               This function will return `None` in v0.23.0 and later.
+        '''
+        thread = threading.Thread(target=self.fade_brightness_thread, args=(finish,), kwargs={
+            'start': start,
+            'interval': interval,
+            'increment': increment,
+            'force': force,
+            'logarithmic': logarithmic,
+            'stoppable': stoppable
+        })
+        thread.start()
+        if blocking:
+            thread.join()
+
+        return self.get_brightness()
+
+    def fade_brightness_thread(
```

Comment:
Not sure there's a use case for users invoking this function directly so it might make more sense as an internal function (eg: `__fade_brightness`). Also means we wouldn't have to maintain a near identical docstring compared to `fade_brightness`

Reply:
Double underscores in Python cause name mangling, which changes the attribute's name internally. This made pytest fail because it couldn't find the attribute. So, I will use a single underscore instead. It allows pytest to access the attribute while still suggesting it's not for direct access.